### ISSUE TITLE
Rename "optimization" to "rewrite"

### ISSUE THREAD

**Issue by brandonwillard** [pymc-devs/Theano-PyMC#238]
What we're naively calling "optimizations" throughout Theano-PyMC are actually "rewrites".  The term "optimization" has some very specific&mdash;and strong&mdash;implications that are **not** met in far too many cases, so we absolutely must correct this sooner than later.

**Comment by canyon289** [comment]
For reference found this talk from 2016 where they explain further what they consider an optimization is 
![Uploading image.png…]()
https://youtu.be/OU8I1oJ9HhI?t=1028

**Comment by brandonwillard** [comment]
> For reference found this talk from 2016 where they explain further what they consider an optimization is
> 
> https://youtu.be/OU8I1oJ9HhI?t=1028

Even with complete awareness of their intention, "rewrite" is still the correct terminology.  

Descriptions like "optimization" can only reasonably be used for the `optdb` classes that collect rewrites of a certain type (e.g. "stabilizations", canonicalization rewrites,  etc.)

**Comment by canyon289** [comment]
Yep, not disagreeing with what you're saying in this issue, just adding a reference
